### PR TITLE
cdk: remove false error logging in _send

### PR DESCRIPTION
**Pull request by alafanechere**
## What
https://github.com/airbytehq/airbyte/pull/11629 introduced a new error log line in the `_send` method which is a false positive error: in the context of it's execution there's no error, the `raise_for_status` is responsible to check if an unexpected error happened.

## How
* Remove the `logger.error` call

### CODE REVIEW

File: airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py [hunk 1]
```diff
@@ -294,7 +294,6 @@ def _send(self, request: requests.PreparedRequest, request_kwargs: Mapping[str,
                 raise DefaultBackoffException(request=request, response=response)
         elif self.raise_on_http_errors:
             # Raise any HTTP exceptions that happened in case there were unexpected ones
-            self.logger.error(f"Request raised an error with response: {response.text}")
             response.raise_for_status()
```
[the remark this answers is not in the document]
Comment (ethanve) [reply airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py:297]: Wouldn't that swallow the error from being caught upstream?

**Comment by grubberr** [comment]
@alafanechere I have merged master branch to your PR and updated version to 0.1.53
because I have used 0.1.52 for my PR
https://github.com/airbytehq/airbyte/pull/11653

**Review by marcosmarxm — APPROVED** [review]
LGTM! thanks @alafanechere

**Pull request merged**